apm-agent-api/src/main/java/co/elastic/apm/api/Transaction.java

```diff
@@ -79,6 +79,48 @@
     @Override
     Transaction addLabel(String key, boolean value);
 
+    /**
+     * Custom context is used to add non-indexed,
+     * custom contextual information to transactions.
+     * Non-indexed means the data is not searchable or aggregatable in Elasticsearch,
+     * and you cannot build dashboards on top of the data.
+     * However, non-indexed information is useful for other reasons,
+     * like providing contextual information to help you quickly debug performance issues or errors.
+     *
+     * @param key   The custom context key.
+     * @param value The custom context value.
+     */
+    @Nonnull
+    Transaction addCustomContext(String key, String value);
```

Comment:
An alternative would be to add a `getContext` method and add the `addCustom()` methods to the `TransactionContext`. Also kinda depends on how we want to make the other context objects accessible. Also, as of 7.x there is no `context` namespace in ES documents anymore. Context only exists in the intake API now. Custom is stored under `transaction.custom`. So maybe we should just name the method `addCustom`?

Reply:
I think it is good as it is. The term `context` is appropriate here regardless of the schema we use.
With `getContext` do you mean we add an API context struct for data storage? Not sure this is necessary.